File: typescript/src/shapes/rectangle.ts
import type { Shape } from "./shape.js";

export class Rectangle implements Shape {
    private readonly height: number; // 将字段设为 private 和 readonly
    private readonly width: number;  // 将字段设为 private 和 readonly

    constructor(height: number, width: number) {
        this.height = height;
        this.width = width;
    }

    getArea(): number {
        return this.height * this.width;
    }

    getHeight(): number {
        return this.height;
    }

    getWidth(): number {
        return this.width;
    }
}

export function newRectangle(height: number, width: number): Rectangle {
    return new Rectangle(height, width);
}

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Rectangle.java
package edu.cmu.cs.cs214.lab02.shapes;

/**
 * Represents a rectangle shape that implements the Shape interface.
 */
public class Rectangle implements Shape {
    private final double height; // 将字段设为 private 和 final
    private final double width;  // 将字段设为 private 和 final

    /**
     * Constructs a Rectangle with the specified height and width.
     *
     * @param height the height of the rectangle
     * @param width the width of the rectangle
     */
    public Rectangle(double height, double width) {
        this.height = height;
        this.width = width;
    }

    /**
     * Returns the area of the rectangle.
     *
     * @return the area of the rectangle
     */
    @Override
    public double getArea() {
        return height * width;
    }

    /**
     * Returns the height of the rectangle.
     *
     * @return the height of the rectangle
     */
    public double getHeight() {
        return height;
    }

    /**
     * Returns the width of the rectangle.
     *
     * @return the width of the rectangle
     */
    public double getWidth() {
        return width;
    }
}
